test/dynamo/test_misc.py

```diff
@@ -925,6 +925,314 @@ def forward(self, arg0_1: "f32[3][1]cpu", arg1_1: "f32[3][1]cpu", arg2_1: "f32[3
             cleanup_op("mylib::foo")
             del lib
 
+    def run_aot_eager(self, f, orig_args):
+        aot_eager_args = pytree.tree_map_only(torch.Tensor, torch.clone, orig_args)
+
+        log_stream, ctx = logs_to_string(
+            "torch._functorch._aot_autograd.dispatch_and_compile_graph", "aot_graphs"
+        )
+
+        result = None
+        with ctx():
+            result = torch.compile(f, backend="aot_eager", fullgraph=True)(
+                *aot_eager_args
+            )
+
+            graph = "\n".join(log_stream.getvalue().strip().split("\n")[4:]).strip()
+        return [aot_eager_args, result, graph]
+
+    def run_inductor(self, f, orig_args):
+        compiled_args = pytree.tree_map_only(torch.Tensor, torch.clone, orig_args)
+
+        log_stream, ctx = logs_to_string(
+            "torch._inductor.compile_fx", "post_grad_graphs"
+        )
+        result = None
+        with ctx():
+            result = torch.compile(f, backend="inductor", fullgraph=True)(
+                *compiled_args
+            )
+
+            graph = "\n".join(log_stream.getvalue().strip().split("\n")[3:]).strip()
+
+        return [compiled_args, result, graph]
+
+    # foo takes two inputs that are not views.
+    def test_auto_functionalize_extra1(self):
+        try:
+            lib = torch.library.Library("mylib", "FRAGMENT")
+            torch.library.define(
+                "mylib::foo",
+                "(Tensor(a!) x, Tensor(b!) y) -> ()",
+                tags=torch.Tag.pt2_compliant_tag,
+                lib=lib,
+            )
+
+            @torch.library.impl("mylib::foo", "cpu", lib=lib)
+            @torch._dynamo.disable
+            def foo_impl(x, y):
+                x.sin_()
+                y.sin_()
+
+            def f(x, y):
+                torch.ops.mylib.foo(x, y)
+                return x + y
+
+            orig_args = (torch.randn(2), torch.randn(2))
+
+            [aot_eager_args, result1, graph_aot] = self.run_aot_eager(f, orig_args)
+            [inductor_args, result2, graph_inductor] = self.run_inductor(f, orig_args)
+            eager_args = pytree.tree_map_only(torch.Tensor, torch.clone, orig_args)
+            result3 = f(*eager_args)
+
+            self.assertEqual(inductor_args, eager_args)
+            self.assertEqual(inductor_args, aot_eager_args)
+
+            self.assertEqual(result3, result1)
+            self.assertEqual(result3, result2)
+
+            if torch._dynamo.config.assume_static_by_default:
+                self.assertExpectedInline(
+                    graph_aot,
+                    """\
+def forward(self, arg0_1: "f32[2][1]cpu", arg1_1: "f32[2][1]cpu"):
+        auto_functionalized = torch.ops.higher_order.auto_functionalized(torch.ops.mylib.foo.default, _x_base_index = 0, _x_size = (2,), _x_stride = (1,), _x_storage_offset = 0, _y_base_index = 1, _y_size = (2,), _y_stride = (1,), _y_storage_offset = 0, _all_bases = [arg1_1, arg0_1])
```

Comment:
In the trivial case (where we are mutating a Tensor instead of the view), can we get auto_functionalized to look better? That is, we shouldn't need to store x_size x_stride x_storage_offset in this case. This happens in the common case and would help convince me that the code works the same the old auto_functionalized.